### ISSUE TITLE
Jedi with PyPy

### ISSUE THREAD

**Issue by davidhalter** [davidhalter/jedi#8]
Running Jedi on PyPy is no problem. (Almost) all tests pass.

But PyPy has a bug in current releases (itertools.tee):
https://bugs.pypy.org/issue1249
It got fixed a few hours ago.

If this fix of PyPy is in the major packaging systems, I will start supporting it (probably PyPy 2.0).

**Comment by brettcannon** [comment]
Could PyPy be tested again?

**Comment by davidhalter** [comment]
I have only recently removed pypy from the tested versions. It was tested for a long time, but allowed to fail. The problem is basically that it generated a lot of errors: https://travis-ci.org/davidhalter/jedi/jobs/277965015.

I have no idea why, but it was definitely not just working. Part of the errors are things that IPython does slightly different (like not having certain dunder methods on certain objects) and other things are probably PyPy bugs. The problem is that too many tests are failing and I would spend way too much time in the PyPy tracker. If someone wants to do that, they can feel free.

On a related note, I realized that PyPy was not really faster than CPython. Rather the contrary was the case: Python was probably like 10-20% faster. This might have changed, but I think that Jedi probably takes too many code paths to really speed it up.

I'm also not quite sure why not even parsing is faster (with parso), but maybe dealing with the stack (list.append, list.pop) is not that much faster.
